tests/constAllocator.cpp

```diff
@@ -0,0 +1,33 @@
+
+/* Copyright (c) 2015-2016, Human Brain Project
+ *                          Daniel@Nachbaur@epf.ch
+ */
+
+#define BOOST_TEST_MODULE constAllocator
+
+#include <boost/test/unit_test.hpp>
+
+#include "serialization.h"
+
+BOOST_AUTO_TEST_CASE(construction)
+{
+    const std::vector< uint8_t > data( 10 );
+    const std::unique_ptr< zerobuf::Allocator > alloc(
+                new zerobuf::ConstAllocator( data.data(), data.size( )));
+
+    BOOST_CHECK_EQUAL( alloc->getData(), data.data( ));
+    BOOST_CHECK_EQUAL( alloc->getSize(), data.size( ));
+    BOOST_CHECK_THROW( alloc->copyBuffer( nullptr, 0 ), std::runtime_error );
+    BOOST_CHECK_THROW( alloc->updateAllocation( 1, 2, 3 ), std::runtime_error );
+    BOOST_CHECK_THROW( alloc->compact( 42 ), std::runtime_error );
+}
+
+BOOST_AUTO_TEST_CASE(create)
+{
+    test::TestSchema object;
+
+    const auto& data = object.toBinary();
+    test::ConstTestSchemaPtr newObject = test::TestSchema::create( data.ptr.get(), data.size );
```

Comment:
The code would look better if the only valid syntax was:

    const test::TestSchema newConstObject = test::TestSchema::fromBinary( data.ptr.get(), data.size );

But doesn't look like something possible in C++.
Still I'd rename it to fromBinary.